spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalogManager.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.spark.catalog;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.client.GravitinoClient;
+import com.datastrato.gravitino.client.GravitinoMetaLake;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** GravitinoCatalogManager is used to sync catalogs from Gravitino server. */
+public class GravitinoCatalogManager {
+  private static final Logger LOG = LoggerFactory.getLogger(GravitinoCatalogManager.class);
+  private static GravitinoCatalogManager gravitinoCatalogManager;
+
+  private volatile boolean isClosed = false;
+  private HashMap<String, Catalog> gravitinoCatalogs = new HashMap<>();
+  private String metalakeName;
+  private GravitinoClient gravitinoClient;
+
+  private GravitinoCatalogManager(String gravitinoUri, String metalakeName) {
+    this.metalakeName = metalakeName;
+    this.gravitinoClient = GravitinoClient.builder(gravitinoUri).build();
+  }
+
+  public static synchronized GravitinoCatalogManager createGravitinoCatalogManager(
+      String gravitinoUrl, String metalakeName) {
+    if (gravitinoCatalogManager != null) {
+      throw new RuntimeException("should not create duplicate GravitinoCatalogManager");
+    }
+    gravitinoCatalogManager = new GravitinoCatalogManager(gravitinoUrl, metalakeName);
+    return gravitinoCatalogManager;
+  }
+
+  public static synchronized GravitinoCatalogManager getGravitinoCatalogManager() {
+    if (gravitinoCatalogManager == null) {
+      throw new RuntimeException("GravitinoCatalogManager has not created yet");
+    }
+    if (gravitinoCatalogManager.isClosed) {
+      throw new RuntimeException("GravitinoCatalogManager " + "is already closed");
+    }
+    return gravitinoCatalogManager;
+  }
+
+  public void close() {
```

Comment:
Should you make this method thread safe?

Reply:
Totally I think there is no need to make create(), get(), close() thread safe, because create is invoked by SparkContext init stage,  get() is inoked in SQL execution stage, close() is invoked in SparkContext finish stage.  this will make code more complex.